pkg/jx/cmd/create_vault.go

```diff
@@ -74,14 +86,31 @@ func NewCmdCreateVault(commonOpts *CommonOptions) *cobra.Command {
 		},
 	}
 
+	// GKE flags
 	cmd.Flags().StringVarP(&options.GKEProjectID, "gke-project-id", "", "", "Google Project ID to use for Vault backend")
 	cmd.Flags().StringVarP(&options.GKEZone, "gke-zone", "", "", "The zone (e.g. us-central1-a) where Vault will store the encrypted data")
+
+	awsCreateVaultOptions(cmd, &options.AWSConfig)
```

Comment:
I am wondering is not cleaner to have two distinct commands:
`jx create vault aws`
`jx create vault gcp`

Reply:
I started implementing that but when realising that one then should check that teamSettings.KubeProvider matches the command made I got cautious. It underlines that splitting into 2 commands goes against the principle that jx shouldn't require the user to give information already known. It might still be the best way, but I'll wait for @jstrachan to pitch in.